code/modules/jobs/job_types/_job.dm

```diff
@@ -133,6 +155,112 @@
 	if(lock_flags || gimmick)
 		SSjob.job_manager_blacklisted |= title
 
+/// Does this job have any space to join?
+/datum/job/proc/has_space()
+	// How many slots does our group have?
+	var/group_slots = get_spawn_position_count(TRUE)
+	// Always available
+	if (group_slots == -1)
+		return TRUE
+	// If the department itself is saturated, return false
+	if (dynamic_spawn_group_multiplier && get_group_position_count() >= group_slots)
+		return FALSE
+	// If the role itself does not care about saturation, return true
+	if (total_positions == -1 || dynamic_spawn_group)
+		return TRUE
+	// If the role itself is saturated, return false
+	if (current_positions >= total_positions  + total_position_delta)
+		return FALSE
+	// Return true otherwise
+	return TRUE
+
+/// Calculates the number of positions currently present in a group
+/// Returns the number of players who are inside a role if the job hasn't reached
+/// its population limit, otherwise groups together roles by checking for their
+/// min_pop_redirect proxy role.
+/datum/job/proc/get_group_position_count()
```

Comment:
This is just my concerning, but due to the similar name with `get_spawn_position_count()`, I believe there can be some misreading or misunderstanding. Not really suggesting to change, but I hope you to consider some better renaming.
in a second thought, I feel `get_spawn_position_count()` might have improper proc name.